catalog/core/catalog-core-impl/pubsub/src/main/java/ddf/catalog/pubsub/PublishedEventHandler.java

```diff
@@ -110,7 +110,7 @@ public void run() {
     private void evaluateSiteBasedSubscription(Set<String> sourceIds, String metacardSourceId) {
       LOGGER.debug(
           "subscription is a site-based subscription starting with site id {}",
-          sourceIds.iterator().next());
+          sourceIds.iterator());
```

Comment:
❗️ Are we tracing an iterator here? That doesn't sound right to me.

Reply:
@paouelle should I just put the `next` back? Maybe this was just a false positive.

Reply:
Based on the log, I would say yes as they were trying to trace only the first one. You will have to surround the trace with an if(LOGGER.isDebugEnabled()) though